workspaces/github-actions/plugins/github-actions/src/components/useDefaultBranch.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useState } from 'react';
+import useAsyncRetry from 'react-use/esm/useAsyncRetry';
+import { githubActionsApiRef } from '../api/GithubActionsApi';
+import { useApi, errorApiRef } from '@backstage/core-plugin-api';
+
+export function useDefaultBranch({
+  hostname,
+  owner,
+  repo,
+}: {
+  hostname?: string;
+  owner: string;
+  repo: string;
+}) {
+  const api = useApi(githubActionsApiRef);
+  const errorApi = useApi(errorApiRef);
+  const [defaultBranch, setDefaultBranch] = useState<string>('');
+
+  const {
+    loading,
+    value: runs,
+    retry,
+    error,
```

Comment:
The `loading` and `error` items here don't seem to be used, I generally will return this to the caller so they can use them as needed/desired.

Reply:
Here's an example of how I've approached this: 

https://github.com/backstage/community-plugins/blob/main/workspaces/azure-devops/plugins/azure-devops/src/hooks/useGitTags.ts